### PR TITLE
introducing physics.particle_shape_and_density

### PR DESCRIPTION
**Pull request by slayoo**

### CODE REVIEW

File: tests/unit_tests/dynamics/collisions/test_sdm_breakup.py [hunk 1]
```diff
@@ -320,22 +320,22 @@ def test_attribute_update_single_breakup(
 
         # Assert
         {
-            "multiplicity": lambda: np.testing.assert_array_equal(
+            "multiplicity": lambda: np.testing.assert_array_almost_equal(
                 particulator.attributes["multiplicity"].to_ndarray(),
```
Comment (abulenok) [inline tests/unit_tests/dynamics/collisions/test_sdm_breakup.py:324]: multiplicity is `int`, so we should check if it is strictly equal

Comment (slayoo) [reply tests/unit_tests/dynamics/collisions/test_sdm_breakup.py:324]: but it fails with equal :(

File: PySDM/attributes/physics/volume.py [hunk 2]
```diff
@@ -1,11 +1,15 @@
 """
-particle (wet) volume, key attribute for coalescence
+particle wet/ice volume (derived from water mass);
```
Comment (abulenok) [inline PySDM/attributes/physics/volume.py:2]:
```suggestion
particle liquid water/ice volume (derived from water mass);
```

File: PySDM/physics/particle_shape_and_density/porous_spheroids.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+"""
+for mixed-phase microphysics as in Shima et al. 2020
```
Comment (abulenok) [inline PySDM/physics/particle_shape_and_density/porous_spheroids.py:2]: maybe add doi? or url?

File: examples/PySDM_examples/deJong_Mackay_et_al_2023/settings_0D.py [hunk 4]
```diff
@@ -37,10 +37,10 @@ def __init__(
         self._steps = [0]
         self.kernel = Geometric()
         self.coal_eff = Berry1967()
-        self.fragmentation = fragmentation or ExponFrag(scale=self.frag_scale)
+        self.fragmentation = fragmentation or Exponential(scale=self.frag_scale)
         self.vmin = 0.0
         self.break_eff = ConstEb(1.0)  # no "bouncing"
-        self.spectrum = Exponential(norm_factor=self.norm_factor, scale=self.X0)
+        self.spectrum = spectra.Exponential(norm_factor=self.norm_factor, scale=self.X0)
```
Comment (abulenok) [inline examples/PySDM_examples/deJong_Mackay_et_al_2023/settings_0D.py:43]: just a nitpick, but I think we should use module names in both cases: `breakup_fragmentations.Exponential` and `spectra.Exponential`

File: tests/smoke_tests/box/dejong_and_mackay_et_al_2023/test_fig_6.py [hunk 5]
```diff
@@ -8,10 +8,10 @@
 from PySDM.backends import CPU, GPU
 from PySDM.dynamics.collisions.breakup_fragmentations import AlwaysN
 from PySDM.dynamics.collisions.coalescence_efficiencies import ConstEc, Straub2010Ec
-from PySDM.physics.constants import si
+from PySDM.physics import si
 
 R_MIN = 0.1 * si.um
-V_MIN = 4 / 3 * np.pi * R_MIN**3
+VMIN = 4 / 3 * np.pi * R_MIN**3
```
Comment (abulenok) [inline tests/smoke_tests/box/dejong_and_mackay_et_al_2023/test_fig_6.py:14]:
```suggestion
```
unused variable

File: tests/smoke_tests/parcel/yang_et_al_2018/test_just_do_it.py [hunk 6]
```diff
@@ -15,7 +15,7 @@
 def test_just_do_it(scheme, adaptive, backend_class=CPU):
```
Comment (abulenok) [inline tests/smoke_tests/parcel/yang_et_al_2018/test_just_do_it.py:15]: what is this test about?

File: tests/unit_tests/dynamics/collisions/test_sdm_breakup.py [hunk 7]
```diff
@@ -759,14 +794,14 @@ def test_nonnegative_even_if_overflow(
...
-        spectrum = Exponential(norm_factor=norm_factor, scale=X0)
+        spectrum = spectra.Exponential(norm_factor=norm_factor, scale=X0)
         attributes = {}
         attributes["volume"], attributes["multiplicity"] = ConstantMultiplicity(
             spectrum
         ).sample(n_sd)
 
         mu = Trivia.volume(const, radius=100 * si.um)
-        fragmentation = ExponFrag(scale=mu)
+        fragmentation = Exponential(scale=mu)
```
Comment (abulenok) [inline tests/unit_tests/dynamics/collisions/test_sdm_breakup.py:804]: same here `breakup_fragmentation.Exponential`

**Review by abulenok — APPROVED** [review]

File: PySDM/physics/particle_shape_and_density/porous_spheroids.py (hunk 3 above)
Comment (abulenok) [reply PySDM/physics/particle_shape_and_density/porous_spheroids.py:2]:
```suggestion
for mixed-phase microphysics as in Shima et al. 2020 (https://doi.org/10.5194/gmd-13-4107-2020)
```

Comment (abulenok) [reply PySDM/physics/particle_shape_and_density/porous_spheroids.py:2]:
```suggestion
for mixed-phase microphysics as in Shima et al. 2020 (https://doi.org/10.5194/gmd-13-4107-2020)
```

File: tests/smoke_tests/parcel/yang_et_al_2018/test_just_do_it.py (hunk 6 above)
Comment (slayoo) [reply tests/smoke_tests/parcel/yang_et_al_2018/test_just_do_it.py:15]: just executes the simulation and does some basic asserts, the aim is to try it with both SciPy and PySDM condenstion solvers (and in future, also with the GPU backend), which is not covered by the example notebook execution

**Pull request merged**